### PR TITLE
add the variables of batch-dml

### PR DESCRIPTION
**Pull request by ekexium**
Signed-off-by: ekexium <eke@fastmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -751,6 +751,33 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
+
+### tidb_batch_insert
+- Scope: Session
+- Persists to cluster: No
+- Type: Boolean
+- Default value: `OFF`
+- This variable is used to control whether to enable the batch insert feature, which is part of the deprecated batch-dml feature. When truned on, insert statements may be split into multiple transactions and be committed non-atomically. You must also turn on `tidb_enable_batch_dml` and set a positive `tidb_dml_batch_size` to make it work.
+
+> **Warning:**
+> batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.
```
Comment (ekexium) [inline system-variables.md:779]: batch-insert cannot be replaced by non-transactional dml for now, but will be.

**Review by cfzjywxk — APPROVED** [review]

File: system-variables.md [hunk 2]
```diff
@@ -751,6 +751,33 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
     In the case of a poor network environment, appropriately increasing the value of this variable can effectively alleviate error reporting to the application end caused by timeout. If the application end wants to receive the error information more quickly, minimize the value of this variable.
 
+### tidb_batch_commit
+- Scope: Session
+- Persists to cluster: No
+- Type: Boolean
+- Default value: `OFF`
+- The variable is used to control whether to enable the deprecated batch commit feature. When truned on, a transaction may be split into multiple transactions by grouping every a few statements and be committed non-atomically. It is not recommended to turn on this feature.
```
Comment (TomShawn) [inline system-variables.md:759]:
```suggestion
### tidb_batch_commit

> **Warning:**
>
> It is **NOT** recommended to enable this variable.

- Scope: Session
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`
- The variable is used to control whether to enable the deprecated batch-commit feature. When this variable is enabled, a transaction might be split into multiple transactions by grouping a few statements and committed non-atomically, which is not recommended.
```

File: system-variables.md [hunk 3]
```diff
@@ -751,6 +751,33 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
+
+### tidb_batch_delete
+- Scope: Session
+- Persists to cluster: No
+- Type: Boolean
+- Default value: `OFF`
+- This variable is used to control whether to enable the batch delete feature, which is part of the deprecated batch-dml feature. When truned on, delete statements may be split into multiple transactions and be committed non-atomically. You must also turn on `tidb_enable_batch_dml` and set a positive `tidb_dml_batch_size` to make it work.
+
+> **Warning:**
+> batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.
```
Comment (TomShawn) [inline system-variables.md:769]:
```suggestion
### tidb_batch_delete

> **Warning:**
>
> This variable is associated with the deprecated batch-dml feature, which might cause data corruption. Therefore, it is not recommended to enable this variable for batch-dml. Instead, use [non-transactional DML](/non-transactional-dml.md).

- Scope: Session
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`
- This variable is used to control whether to enable the batch-delete feature, which is a part of the deprecated batch-dml feature. When this variable is enabled, `DELETE` statements might be split into multiple transactions and committed non-atomically. To make it work, you also need to enable `tidb_enable_batch_dml` and set a positive value for `tidb_dml_batch_size`, which is not recommended.
```

File: system-variables.md [hunk 4]
```diff
@@ -751,6 +751,33 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
+
+### tidb_batch_insert
+- Scope: Session
+- Persists to cluster: No
+- Type: Boolean
+- Default value: `OFF`
+- This variable is used to control whether to enable the batch insert feature, which is part of the deprecated batch-dml feature. When truned on, insert statements may be split into multiple transactions and be committed non-atomically. You must also turn on `tidb_enable_batch_dml` and set a positive `tidb_dml_batch_size` to make it work.
+
+> **Warning:**
+> batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.
```
Comment (TomShawn) [inline system-variables.md:779]:
```suggestion
### tidb_batch_insert


> **Warning:**
>
> This variable is associated with the deprecated batch-dml feature, which might cause data corruption. Therefore, it is not recommended to enable this variable for batch-dml. Instead, use [non-transactional DML](/non-transactional-dml.md).

- Scope: Session
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`
- This variable is used to control whether to enable the batch-insert feature, which is a part of the deprecated batch-dml feature. When this variable is enabled, `INSERT` statements might be split into multiple transactions and committed non-atomically. To make it work, you also need to enable `tidb_enable_batch_dml` and set a positive for `tidb_dml_batch_size`, which is not recommended.
```

File: system-variables.md [hunk 5]
```diff
@@ -1088,6 +1115,10 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Unit: Rows
 - When this value is greater than `0`, TiDB will batch commit statements such as `INSERT` or `LOAD DATA` into smaller transactions. This reduces memory usage and helps ensure that the `txn-total-size-limit` is not reached by bulk modifications.
 - Only the value `0` provides ACID compliance. Setting this to any other value will break the atomicity and isolation guarantees of TiDB.
+- You must also turn on `tidb_enable_batch_dml` and at least one of `tidb_batch_insert` and `tidb_batch_delete` to use this feature.
```
Comment (TomShawn) [inline system-variables.md:1118]:
```suggestion
- To make this variable work, you also need to enable `tidb_enable_batch_dml` and at least one of `tidb_batch_insert` and `tidb_batch_delete`.
```

File: system-variables.md [hunk 6]
```diff
@@ -1088,6 +1115,10 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
+
+> **Warning:**
+> batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.
```
Comment (TomShawn) [inline system-variables.md:1121]: It is recommended to move this warning to line 1109.
```suggestion
```

File: system-variables.md [hunk 7]
```diff
@@ -1163,6 +1194,16 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
 
+### tidb_enable_batch_dml
+- Scope: GLOBAL
+- Persists to cluster: Yes
+- Type: Boolean
+- Default values: `OFF`
+- This variable contronls whether to enable the deprecated batch-dml feature. When it's enabled, certain statements may be split into multiple transactions, which means it's non-atomic and should be used with care. When using the feature you must ensure there is no concurrent opertaions on the data you are operating on. You must also specify a positive `tidb_batch_dml_size` and turn on at least one of `tidb_batch_insert` and `tidb_batch_delete` to make it work.
+
+> **Warning:**
+> batch-dml is deprecated because it's easy to misuse it and cause data corruption. Consider using [non-transactional DML](/non-transactional-dml.md) instead.
```
Comment (TomShawn) [inline system-variables.md:1205]:
```suggestion
### tidb_enable_batch_dml

> **Warning:**
>
> This variable is associated with the deprecated batch-dml feature, which might cause data corruption. Therefore, it is not recommended to enable this variable for batch-dml. Instead, use [non-transactional DML](/non-transactional-dml.md).

- Scope: GLOBAL
- Persists to cluster: Yes
- Type: Boolean
- Default values: `OFF`
- This variable controls whether to enable the deprecated batch-dml feature. When it is enabled, certain statements might be split into multiple transactions, which is non-atomic and should be used with care. When using batch-dml, you must ensure that there are no concurrent operations on the data you are operating on. To make it work, you must also specify a positive value for `tidb_batch_dml_size` and enable at least one of `tidb_batch_insert` and `tidb_batch_delete`.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**